### PR TITLE
[GSS]Unable to expand ocs-storagecluster-ceph-rbd PVCs provisioned in Filesystem mode

### PR DESCRIPTION
**Pull request by Shrivaibavi**
bz - https://bugzilla.redhat.com/show_bug.cgi?id=2164617

### CODE REVIEW

File: tests/manage/pv_services/pvc_resize/test_pod_deletion_during_pvc_expansion.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+                f" on pod {rbd_pod.name}. New size on mount is not "
+                f"{pvc_size_new}G as expected, but {new_size_mount}. "
+                f"Checking again."
+            )
+        log.info(
+            f"Verified: Modified size {pvc_size_new}G is reflected " f"on all pods."
+        )
+
+        # Respin app-pods multiple times
+        for i in range(1, 5):
```
Comment (PrasadDesala) [inline tests/manage/pv_services/pvc_resize/test_pod_deletion_during_pvc_expansion.py:118]: Can we increase the iteration count here? Maybe to 15. I think the loop doesn't take much time as the only operation that we are doing in the loop is delete.

File: tests/manage/pv_services/pvc_resize/test_pod_deletion_during_pvc_expansion.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+@skipif_ocs_version("<4.5")
+@skipif_upgraded_from(["4.4"])
+class TestPodRespinDuringPvcExpansion(ManageTest):
+    """
+    Tests to verify PVC expansion during app pod respins
+
+    """
+
+    @pytest.fixture()
+    def create_pod_pvc(self, request, pvc_factory, service_account_factory):
```
[the remark this answers is not in the document]
Comment (Shrivaibavi) [reply tests/manage/pv_services/pvc_resize/test_pod_deletion_during_pvc_expansion.py]: Changed

File: tests/manage/pv_services/pvc_resize/test_pod_deletion_during_pvc_expansion.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+)
+from ocs_ci.helpers import helpers
+from ocs_ci.framework import config
+
+log = logging.getLogger(__name__)
+
+
+@tier4c
+@skipif_ocs_version("<4.5")
+@skipif_upgraded_from(["4.4"])
```
[the remark this answers is not in the document]
Comment (Shrivaibavi) [reply tests/manage/pv_services/pvc_resize/test_pod_deletion_during_pvc_expansion.py]: Removed

File: tests/manage/pv_services/pvc_resize/test_pod_deletion_during_pvc_expansion.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        Verify PVC expansion during rbd-app pod respins
+
+        """
+
+        rbd_pvc, rbd_pod = create_pod_pvc
+
+        if config.ENV_DATA["platform"].lower() in constants.MANAGED_SERVICE_PLATFORMS:
+            pvc_size_new = 15
+        else:
+            pvc_size_new = 25
```
[the remark this answers is not in the document]
Comment (Shrivaibavi) [reply tests/manage/pv_services/pvc_resize/test_pod_deletion_during_pvc_expansion.py]: Incorporated in this commit https://github.com/red-hat-storage/ocs-ci/pull/3586, Hope removing this wont cause any limitation issues in MS

File: tests/manage/pv_services/pvc_resize/test_pod_deletion_during_pvc_expansion.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        * Create RBD Filesystem PVC
+        * Mount it to the application pod
+        * Expand the PVC
+        * Restart the pod multiple times
+        * Check csi-rbdplugin pod logs on the node where the app pod is running for error
+            'Internal desc = failed to get device for stagingtarget path'
+        * The expansion should be successful, and the app should be running
+
+        """
+        rbd_pod = dc_pod_factory(size=10)
```
Comment (PrasadDesala) [inline tests/manage/pv_services/pvc_resize/test_pod_deletion_during_pvc_expansion.py:46]: this test is valid for external mode too. Resource name changes in external mode, please check if it is handled in this fixture

Comment (PrasadDesala) [reply tests/manage/pv_services/pvc_resize/test_pod_deletion_during_pvc_expansion.py:46]: Looks like the dc_pod_factory handles it, resolving the comment.

**Review by PrasadDesala — APPROVED** [review]

**Review by jilju — APPROVED** [review]

**Pull request merged**